dev_docs/Troubleshooting_Guide.rst

```diff
@@ -0,0 +1,116 @@
+  File "/home/vagrant/git/st2/st2common/st2common/models/db/__init__.py", line 207, in insert
+    instance = self.model.objects.insert(instance)
+  File "/home/vagrant/git/st2/virtualenv/local/lib/python2.7/site-packages/mongoengine/queryset/base.py", line 307, in insert
+    raise NotUniqueError(message % unicode(err))
+  NotUniqueError: Could not save document (E11000 duplicate key error index: st2.role_d_b.$name_1 dup key: { : "admin" })
+  2016-04-26 18:27:15,606 140317722756912 DEBUG base [-] Conflict while trying to save in DB.
+  Traceback (most recent call last):
+    File "/home/vagrant/git/st2/st2common/st2common/persistence/base.py", line 120, in insert
+    model_object = cls._get_impl().insert(model_object)
+  File "/home/vagrant/git/st2/st2common/st2common/models/db/__init__.py", line 207, in insert
+    instance = self.model.objects.insert(instance)
+  File "/home/vagrant/git/st2/virtualenv/local/lib/python2.7/site-packages/mongoengine/queryset/base.py", line 307, in insert
+    raise NotUniqueError(message % unicode(err))
+  NotUniqueError: Could not save document (E11000 duplicate key error index: st2.role_d_b.$name_1 dup key: { : "observer" })
+  2016-04-26 18:27:15,607 140317722756912 DEBUG base [-] Conflict while trying to save in DB.
+  Traceback (most recent call last):
+    File "/home/vagrant/git/st2/st2common/st2common/persistence/base.py", line 120, in insert
+      model_object = cls._get_impl().insert(model_object)
+    File "/home/vagrant/git/st2/st2common/st2common/models/db/__init__.py", line 207, in insert
+     instance = self.model.objects.insert(instance)
+    File "/home/vagrant/git/st2/virtualenv/local/lib/python2.7/site-packages/mongoengine/queryset/base.py", line 307, in insert
+      raise NotUniqueError(message % unicode(err))
+  NotUniqueError: Could not save document (E11000 duplicate key error index: st2.role_d_b.$name_1 dup key: { : "system_admin" })
+  2016-04-26 18:27:15,676 140317722756912 INFO driver [-] Generating grammar tables from /usr/lib/python2.7/lib2to3/Grammar.txt
+  2016-04-26 18:27:15,693 140317722756912 INFO driver [-] Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt
+  
+- To figure out whats wrong let us dig down further. Activate the virtualenv in st2 and run following command :
+
+.. code:: bash
+
+   (virtualenv) $ ST2_CONFIG_PATH=conf/st2.dev.conf ./virtualenv/bin/gunicorn_pecan ./st2api/st2api/gunicorn_config.py -k eventlet -b 0.0.0.0:9101 --workers 1
+
+The above mentioned command will give out logs, we may find some error in the end of logs like this:
+
+.. code:: bash
+
+    File "/home/vagrant/git/st2/st2common/st2common/models/api/keyvalue.py", line 19, in <module>
+      from keyczar.keys import AesKey
+  ImportError: No module named keyczar.keys
+  
+So the problem is : module keyczar is missing. This module can be downloaded using following command:
+
+*Solution:*
+
+.. code:: bash
+
+  (virtualenv) $ pip install python-keyczar
```

Comment:
Usually in situation like this, the right step would be to run `make requirements` or `pip install -r requirements.txt`.

This way it will use the correct version we want which is specified in the requirements file.